bftengine/tests/bcstatetransfer/test_app_state.hpp

```diff
@@ -77,16 +81,17 @@ class TestAppState : public IAppState {
     memcpy(&bl.block, block, blockSize);
     bl.actualSize = blockSize;
     last_block_ = blockId;
+    blocks_.emplace(blockId, bl);
     return true;
   }
 
-  // TODO - implement
   std::future<bool> putBlockAsync(uint64_t blockId,
                                   const char* block,
                                   const uint32_t blockSize,
                                   bool trylinkSTChainFrom = true) override {
-    ConcordAssert(false);
-    return std::async([]() { return false; });
+    std::future<bool> future =
+        std::async(std::launch::async, [&]() { return putBlock(blockId, block, blockSize, trylinkSTChainFrom); });
```

Comment:
Can you please rename all trylinkSTChainFrom to lastBlock? (only in this file)
This is a recent change, too bad I didn't tell you about it.